### PR TITLE
Fix bootstrap by SIM authentication

### PR DESCRIPTION
**Pull request by c9katayama**
This PR fixes bootstrap issue by SIM authentication.

- Since Leshan 1.2, Server object is necessary for bootstrapping with SIM authentication, so fixed to return the Server object even if there is no credential.
- Fixed the problem of NPE when loading credentials.
- Set recommended lifetime value to server object when bootstrap is succeeded

### CODE REVIEW

File: soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java [hunk 1]
```diff
@@ -322,10 +322,8 @@ private void initServer(ObjectsInitializer initializer, Credentials credentials)
 		if (objectInstanceMap.containsKey(LwM2mId.SERVER)) {
 			return;
 		}
-		if (credentials != null) {
-			serverObject = InventoryAgentHelper.getServerInfo(credentials);
-			initializer.setInstancesForObject(LwM2mId.SERVER, serverObject);
-		}
```
Comment (moznion) [inline soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java:328]: what intention is this null check removing?

File: soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java [hunk 2]
```diff
@@ -285,15 +285,15 @@ private Credentials loadCredentials() {
...
-	private long calculateLifetimeSec() {
-		int defaultLifetimeSec = 60;
-		// according to
-		// org.eclipse.leshan.client.californium.CaliforniumEndpointsManager.java#getMaxCommunicationPeriodFor
-		return defaultLifetimeSec + 247 + 30;
+	private void setRecommendedLifetime(Credentials credentials) {
+		if (credentials != null) {
+			long recommendedLifetime = InventoryAgentHelper.calculateRecommendedLifetimeSec(credentials.getLifetime());
+			credentials.setLifetime(recommendedLifetime);
+		}
```
Comment (moznion) [inline soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java:296]: It seems this makes sense, but once the `credentials` was null, there is no chance to set the lifetime property?

Comment (c9katayama) [reply soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java:296]: Yes, if the credentials is null, bootstrap is happen to get credentials from bootstrap server. The return value of the bootstrap request contains default lifetime value, and the client tries to set recommended lifetime value after bootstrap succeeds.
https://github.com/soracom/soracom-inventory-agent-for-java/pull/14/files/706928bb77e5ab315e28df642835f58343c3a3e2#diff-0185eee4d952255fa6beb44a27fb284b091a5e83aa4c9e8e170591c07c345a9cR76

File: soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentHelper.java [hunk 3]
```diff
@@ -63,12 +64,26 @@ public static Security getSecurityInfoForBootstrap(String bootstrapServerURI, by
...
 		} else {
 			return new Server(c.getShortServerId(), c.getLifetime(), c.getBindingMode(), c.isNotifyWhenDisable());
 		}
 	}
 
+	public static long calculateRecommendedLifetimeSec(long lifetimeSec) {
+		return Math.max(lifetimeSec, calculateDefaultLifetimeSec());
+	}
+
+	public static long calculateDefaultLifetimeSec() {
```
[the remark this answers is not in the document]
Comment (c9katayama) [reply soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentHelper.java]: OK, fixed it.

File: soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java (hunk 1 above)
Comment (c9katayama) [reply soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java:328]: I updated InventoryAgentHelper.getServerInfo() to return Server object even credentials is null, so I needed to remove the null check.
https://github.com/soracom/soracom-inventory-agent-for-java/pull/14/files/706928bb77e5ab315e28df642835f58343c3a3e2#diff-2d9f8e0b00b3293c40ed7cb2973ee656e4d3f2bc288a6b231fac79059d085c30R65

**Review by moznion — APPROVED** [review]
Okay, thanks for the information and adjustment.
I appreciate your work! Let's ship it.

**Pull request merged**